src/map.rs

```diff
@@ -127,7 +127,7 @@ pub struct Server<T> {
     map_svc: Option<T>,
 }
 
-impl<T> Server<T> {
+impl<T: Mapper> Server<T> {
```

Comment:
In Rust, usually trait bounds are specified at the locations where it is needed. Here, it would be the `start` and `start_with_shutdown` methods. If we are moving the trait bounds to full implementation block itself, let's remove it from the `where` clause of those methods and make `T: Mapper + Send + Sync + 'static`